packages/scandipwa/src/component/NewsletterSubscription/NewsletterSubscription.container.js

```diff
@@ -97,12 +98,13 @@ export class NewsletterSubscriptionContainer extends PureComponent {
             <NewsletterSubscription
               { ...this.containerProps() }
               { ...this.containerFunctions }
+              key={ window.location.pathname }
             />
         );
     }
 }
 
-export default connect(
+export default withRouter(connect(
```

Comment:
If you take pathname from window, there is no need to wrap container into `withRouter` (withRouter will provide access to such props as `history` and `location`, but I don't see that them to be used in this case).

Reply:
I tried it, the solution seems not to work without withRouter. I think it has something to do with passing new props every time user changes page, which then refreshes the component.

Reply:
It happens due to no prop being updated, if `withRouter` is not used => nothing to trigger update in component. In this case it's better to use `withRouter` + replace `window.location.pathname` with `this.props.location.pathname` (with props destructuring).